robot-command/src/main/java/org/obolibrary/robot/AnnotateCommand.java

```diff
@@ -75,6 +75,7 @@ public AnnotateCommand() {
         "annotate-defined-by",
         true,
         "if true, annotate all entities in the ontology with the ontology IRI");
+    o.addOption("e", "interpolate", false, "enable interpolation within annotation values");
```

Comment:
I would personally prefer if the "interpolation" flag would be computed dynamically like so (pseudocode):

```
boolean interpolate = value.contains("{%");
```

This is a theoretically breaking change but I cannot imagine that anyone in the world of `robot annotate` has ever used the combination of `%{` (let alone `%{onotlogy_iri}` if you want to play it unnecessarily safe) in an annotation value.

A slightly less destructive option would be:

```
 boolean interpolate = CommandLineHelper.getBooleanValue(line, "interpolate", true, true);
```

So people could, in an emergency, switch of interpolation (but its used by default).

Reply:
> I would personally prefer if the "interpolation" flag would be computed dynamically > like so (pseudocode):
> 
> boolean interpolate = value.contains("%{");

That would mean that we search through the value twice: a first time just to decide whether we need to interpolate, and a second time to do the actual interpolation.

We might as well try to perform interpolation unconditionally then – and if the value happens _not_ to contain a placeholder, then nothing will be interpolated.

Reply:
> I cannot imagine that anyone in the world of robot annotate has ever used the combination of %{

Well, I can imagine that.

I can imagine a scenario where someone hacked their own interpolation mechanism downstream of ROBOT. That is, they use ROBOT to inject `%{...}` placeholders into annotations, and then do a post-processing step using some custom scripts to replace those placeholders with the actual values they need. Such a custom pipeline would break if ROBOT suddenly does its own interpolation.

Far-fetched? Maybe, or maybe not.

[Mandatory XKCD](https://xkcd.com/1172)

Reply:
Sorry @matentzn, I want to keep the explicit behaviour in the current version that requires `--interpolate true`. I hope that being explicit will (1) avoid surprises for existing workflows, and (2) encourage confused users to look up "interpolate" in the docs if they see strange glyphs in the command.